### PR TITLE
PSScriptAnalyzer and VS Code tooling

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
#### Description

This adds the following files:

* `launch.json` - used by vscode in order to debug PowerShellForGitHub with ease
* `PSScriptAnalyzerSettings.psd1` - used by PSSA to check syntax and also by VS Code to show problems for all rules

#### Issues Fixed

N/A

#### References

PSSA Rules: https://github.com/PowerShell/PSScriptAnalyzer/tree/master/RuleDocumentation
Specifically: https://github.com/PowerShell/PSScriptAnalyzer/blob/master/RuleDocumentation/UseCompatibleSyntax.md

#### Checklist

- [x] You actually ran the code that you just wrote, especially if you did just "one last quick change".
- [x] Comment-based help added/updated, including examples.
- [x] [Static analysis](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis)
is reporting back clean.
- [x] New/changed code adheres to our [coding guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#coding-guidelines).
- [x] Changes to the manifest file follow the [manifest guidance](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#module-manifest).
- [x] Unit tests were added/updated and are all passing. See [testing guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#testing).
- [x] Relevant usage examples have been added/updated in [USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md).
- [x] If desired, ensure your name is added to our [Contributors list](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#contributors)

### CODE REVIEW

File: PSScriptAnalyzerSettings.psd1 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# PSScriptAnalyzerSettings.psd1
+# Settings for PSScriptAnalyzer invocation.
+# All default rules are also enabled.
+@{
+    Rules = @{
+        PSUseCompatibleSyntax = @{
```
Comment (rjmholt) [inline PSScriptAnalyzerSettings.psd1:6]: Cool!

File: build/pipelines/templates/run-staticAnalysis.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ steps:
     displayName: 'Install PSScriptAnalyzer'
 
   - powershell: |
-       $results = Invoke-ScriptAnalyzer -Path ./ –Recurse
+       $results = Invoke-ScriptAnalyzer -Settings ./PSScriptAnalyzerSettings.psd1 -Path ./ –Recurse
```
Comment (HowardWolosky) [inline build/pipelines/templates/run-staticAnalysis.yaml:12]: Can you make a similar change to CONTRIBUTING.md as well?

https://github.com/microsoft/PowerShellForGitHub/blob/742b85324e94cfc1b3ba14edc8a450cccdc882fa/CONTRIBUTING.md#L152-L157

Comment (TylerLeonhardt) [reply build/pipelines/templates/run-staticAnalysis.yaml:12]: Done. + another reference

**Comment by HowardWolosky** [comment]
@TylerLeonhardt  - Sorry for the delay on getting this merged.  If you can get the merge conflict resolved, I'll get this merged in.  Thanks again for your efforts here.

**Comment by TylerLeonhardt** [comment]
done

**Pull request merged**